sqlitecluster/SQLiteNode.cpp

```diff
@@ -2494,12 +2548,14 @@ void SQLiteNode::postPoll(fd_map& fdm, uint64_t& nextActivity) {
             int messageSize = message.deserialize(socket->recvBuffer);
             if (messageSize) {
                 socket->recvBuffer.consumeFront(messageSize);
-                if (SIEquals(message.methodLine, "NODE_LOGIN")) {
+                // Allow either LOGIN or NODE_LOGIN until we deprecate NODE_LOGIN.
+                if (SIEquals(message.methodLine, "NODE_LOGIN") || SIEquals(message.methodLine, "LOGIN")) {
                     SQLitePeer* peer = getPeerByName(message["Name"]);
                     if (peer) {
                         if (peer->setSocket(socket)) {
-                            _sendPING(peer);
                             _onConnect(peer);
+                            _sendPING(peer);
+                            _onMESSAGE(peer, message);
```

Comment:
I don't know that this is a problem, but I don't see this line before -- it feels like an important line.  Just double checking this is intentional?

Reply:
It is important and intentional. The idea is that when we receive a LOGIN (as opposed to a NODE_LOGIN) we will process it immediately, and mark the node as forked in the very first message we receive.

This is also the reason we added a "do nothing" block here: 
https://github.com/Expensify/Bedrock/pull/1999/files#diff-2d1c5f24a4b7ebc3076630d8ee3a1155ad5e7789abe72cff977e615541c319d6R1272

Because we did *not* call this before, _onMESSAGE had no handler for NODE_LOGIN, and if it were to get called, it would throw an exception. With the "do nothing" change above, we can handle either LOGIN or NODE_LOGIN here just fine, until we rip out NODE_LOGIN next week. Only a LOGIN as the first message will trigger the immediate fork detection, NODE_LOGIN does not supply a hash.